.github/workflows/release.yml

```diff
@@ -0,0 +1,37 @@
+name: Release
+on:
+  push:
+    tags:        
+      - 'v*'
+jobs:
+  test:
+    uses: ./.github/workflows/test.yml
+  npm-publish:
+    needs: test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14
+          cache: 'npm'
+      - run: npm ci
+      - run: npm run build
+      # Bump the version in package.json
+      # https://docs.github.com/en/actions/learn-github-actions/contexts#github-context
+      - run: npm version ${{ github.ref_name }}
+      - run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+      # After publish, open a PR to update the version in package.json
+      - uses: peter-evans/create-pull-request@v4
+        with:
+          title: Bump package.json version to ${{ github.ref_name }}
+          body: |
+            Bump package.json version to ${{ github.ref_name }}
+          branch: npm-version-bump
+          base: master
+          labels: |
+            dependencies
+            version-bump
```

Comment:
It feels weird to create a PR for this. Especially since we should probably update `package.json` before tagging a new version, rather than after.

Reply:
Thinking more about this, this seems outright wrong to me. The published NPM package's `package.json` needs to match the version number passed to `npm version`, which implies that `package.json` _must_ be updated before tagging.

Reply:
This flow here is designed with the original goal of making the whole process as simple and automated as possible, so with that in mind I set it up like this to keep everything simple.

Instead of having to bump the version in the package.json yourself (which I wasn't sure if it was being maintained or not), you simply create a GitHub release.
The CI then checks out the repo locally, and uses `npm version` to locally bump the version in the package.json to the same number as the tag that was created using the github release.
Then, in the next step, when npm publish runs, locally the version number in the package.json is bumped, so it'll publish that new version

However, this commit is only local in the CI machine, so to keep the repo in sync, it then opens a PR with the change to the version number in the package.json

Reply:
This can be changed of course, but the alternative to this is a user manually bumping the version on their local machine, opening a PR, merging it, then creating a release with that exact same version

It's a lot more manual steps with room for error for the same outcome

Reply:
Oh!!! I think I understand what you mean

While the published npm version will have the correct version number, the git tag/release's package.json will still technically be on the previous version

I can see why that would be undesirable, if not outright wrong

If you want I can change the whole flow so that instead of being triggered on release, it triggers on commit

If a commit to the master branch bumps the package.json version, that will then be published to npm via `npm publish`

I can also add a separate super simple workflow that is triggered manually via the UI, where from a drop-down you simply select the version to bump (major, minor, patch) with a single click and it'll automatically create a PR to update the version